lib/middleware/crawler_hooks.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module Middleware
+  class CrawlerHooks
+    def initialize(app)
+      @app = app
+    end
+
+    def call(env)
+      request = Rack::Request.new(env)
+      status, headers, response = @app.call(env)
+
+      if status == 200 && headers["Content-Type"]&.include?("text/html") &&
+           CrawlerDetection.crawler?(request.user_agent, request.get_header("HTTP_VIA"))
+        response = transform_response(request:, response:)
+      end
+
+      [status, headers, response]
+    end
+
+    private
+
+    def transform_response(request:, response:)
+      locale = request.params[Discourse::LOCALE_PARAM]
+
+      if SiteSetting.content_localization_enabled &&
+           SiteSetting.content_localization_crawler_param && locale.present?
```

Comment:
We might want to consider moving the site settings guards before the crawler check method since the crawler checks are unnecessary when the site settings are disabled.